core/src/providers/provider.rs

```diff
@@ -76,7 +76,11 @@ impl std::fmt::Display for ModelError {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         write!(
             f,
-            "[model_error(retryable={})] {}",
+            "[model_error(retryable={}{})] {}",
```

Comment:
This one adds the request_id to the user visible model_error rendering. Might be a bit too much we'll see as we get these in the UI (behind view error in the conversation)

Reply:
We could consider adding the run_id here as well